src/main/java/com/microsoft/azure/relay/HybridConnectionListener.java

```diff
@@ -627,8 +627,8 @@ else if (command.getRenewToken() != null) {
 							RelayLogger.logEvent("sendCommand", this, json);
 							return webSocket.writeAsync(json, timeout, WriteMode.TEXT)
 								.thenCompose($void -> {
-									if (buffer != null) {
-										return webSocket.writeAsync(buffer.array());
```

Comment:
>array [](start = 45, length = 5)

Was the problem that the ByteBuffer.array might include extra data at the end?

Reply:
The switch to ByteArrayOutputStream is mostly because it's self-resizable. If the user writes multiple times, it can easily lead to exceptions if the ByteBuffer wasn't set to the right size in the beginning.

---
In reply to: [276023404](https://github.com/Azure/azure-relay-java/pull/39#discussion_r276023404) [](ancestors = 276023404)